lib/data/locationConstraintParser.js

```diff
@@ -32,7 +33,40 @@ function parseLC() {
             clients[location].clientType = 'scality';
         }
         if (locationObj.type === 'aws_s3') {
-            // TODO
+            let s3Params;
+            // users can either include the desired profile name from their
+            // ~/.aws/credentials file or include the accessKeyId and
+            // secretAccessKey directly in the locationConfig
+            if (locationObj.details.credentialsProfile) {
+                s3Params = {
+                    endpoint: `https://${locationObj.details.awsEndpoint}`,
```

Comment:
I'd really rather not repeat these items.  can you build s3Params and then just add on the credentials or accessKeyId/secretAccessKey params as applicable?